src/app/centers/createcenter/createcenter.component.ts

```diff
@@ -0,0 +1,174 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { FormBuilder, FormGroup, Validators, FormControl } from '@angular/forms';
+
+/** Custom Services */
+import { GroupsService } from 'app/groups/groups.service';
+import { CentersService } from '../centers.service';
+
+@Component({
+  selector: 'mifosx-createcenter',
+  templateUrl: './createcenter.component.html',
+  styleUrls: ['./createcenter.component.scss']
+})
+export class CreatecenterComponent implements OnInit {
+
+  /** Minimum closing date allowed. */
+  minDate = new Date(2000, 0, 1);
+  /** Maximum closing date allowed. */
+  maxDate = new Date();
+  /** Center form. */
+  centerForm: FormGroup;
+  /** Office data. */
+  officeData: any;
+  /** Group data. */
+  groupData: any;
+  /** Group detail */
+  groupDataDetail: any[] = [];
+  value: Boolean;
+  /** Staff data. */
+  staffData: any;
+  disableSelect = new FormControl(false);
+  /**  date  form control. */
+  date = new FormControl(new Date());
+  serializedDate = new FormControl((new Date()).toISOString());
+
+  choice: any[] = [];
+  /** Group Members. */
+  groupMembers: any[] = [];
+
+  constructor(private formBuilder: FormBuilder,
+    private centerService: CentersService,
+    private groupService: GroupsService,
+    private route: ActivatedRoute,
+    private router: Router) {
+    this.route.data.subscribe((data: { offices: any }) => {
+      this.officeData = data.offices;
+    });
+  }
+
+  ngOnInit() {
+    this.createCenterForm();
+    this.centerForm.controls['activationDate'].disable();
+    this.value = true;
+  }
+
+  /**
+   * Creates the center  form.
+   */
+  createCenterForm() {
+    this.centerForm = this.formBuilder.group({
+      'name': ['', Validators.required],
+      'officeId': ['', Validators.required],
+      'staffId': [''],
+      'groupMembers': [''],
+      'externalId': [''],
+      'submittedOnDate': ['', Validators.required],
+      'activationDate': ['', Validators.required],
+    });
+  }
+
+  onclick(id: number) {
+    /**
+     * Gets the Group for a specific office .
+     */
+    this.groupService.getGroupByOfficeId(id + 1)
+      .subscribe(
+        (data => {
+          this.groupData = data;
+        })
+      );
+
+    /**
+     * Gets the Staff for a specific office .
+     */
+    this.centerService.getStaff(id + 1).subscribe((data: {
+      officeId: any,
+      active: any,
+      officeOptions: any,
+      staffOptions: any,
+    }) => {
+      this.staffData = data.staffOptions;
+    });
+
+  }
+
+  /**
+   * Enable and disable Activation date .
+   */
+  disable() {
```

Comment:
Use value changes on the form as mentioned in the comments within the html file.